python/paddle/v2/framework/framework.py

```diff
@@ -538,5 +538,20 @@ def __init__(self, block, shape, dtype, **kwargs):
 
 
 # program is a global instance.
-g_program = Program()
-g_init_program = Program()
+g_program_dict = dict()
+
+
+def switch_g_program(prog, init_prog):
```

Comment:
Is this API supposed to exposed to PaddlePaddle users?  It might be dangerous to expose the fact that we are having two programs -- the main one and the initializer -- to the user at this moment.

I think it is reasonable to have the main program and the initializer program -- the former is like the `main` function in C/C++, and the latter the C/C++ runtime entry point that initializes the global variables.  It is just that we might expose only the main program to the users.